eta/core/utils.py

```diff
@@ -1646,32 +1646,34 @@ def hash(path):
             return str(hashlib.md5(f.read()).hexdigest())
 
 
-def make_temp_dir(dir=None):
+def make_temp_dir(basedir=None):
     '''Makes a temporary directory.
 
     Args:
-        dir: an optional directory in which to put the temp dir
+        basedir: an optional directory in which to create the new directory
 
     Returns:
         the path to the temporary directory
     '''
-    return tempfile.mkdtemp(dir=dir)
+    if basedir:
+        ensure_dir(basedir)
```

Comment:
This actually isn't going to fix the problem. If `/tmp` doesn't exist, then:
```
tempfile.mkdtemp()
```
will fail

Reply:
what? `make_temp_dir("/tmp")` will make a directory `"/tmp/xxx"`, so the first step is to ensure that `"/tmp"` exists